extensions/sockets/src/main/java/org/mule/module/socket/api/connection/udp/AbstractUdpConnection.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.api.connection.udp;
+
+import static java.lang.String.format;
+import static org.mule.module.socket.internal.SocketUtils.getAddress;
+import org.mule.module.socket.api.connection.AbstractSocketConnection;
+import org.mule.module.socket.api.socket.udp.UdpSocketProperties;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.core.api.serialization.DefaultObjectSerializer;
+import org.mule.runtime.core.api.serialization.ObjectSerializer;
+
+import java.net.DatagramSocket;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides fields and behaviour common to {@link }
+ */
+public abstract class AbstractUdpConnection extends AbstractSocketConnection
+{
+
+    protected DatagramSocket socket;
+    protected final UdpSocketProperties socketProperties;
+    private final Logger LOGGER = LoggerFactory.getLogger(AbstractUdpConnection.class);
+
+    @DefaultObjectSerializer
+    @Inject
+    protected ObjectSerializer objectSerializer;
+
+    public AbstractUdpConnection(UdpSocketProperties socketProperties, String host, Integer port) throws ConnectionException
+    {
+        super(host, port);
+        this.socketProperties = socketProperties;
+    }
+
+    @Override
+    public void disconnect()
+    {
+        LOGGER.debug("Closing UDP socket");
+        socket.close();
+    }
+
+
+    @Override
+    public void validate() throws ConnectionException
+    {
+        if (socket.isClosed())
+        {
+            throw new ConnectionException("Requester UDP socket is closed");
```

Comment:
a validate method should not throw a ConnectionException. It should inform a ConnectionValidationResult and let the invoker decide what to do with it.